src/Encamina.Enmarcha.SemanticKernel.Abstractions/ILengthFunctions.cs

```diff
@@ -30,6 +32,41 @@ public interface ILengthFunctions : AI.Abstractions.ILengthFunctions
     /// <seealso href="https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb"/>
     public static Func<string, string, int> LengthByTokenCountUsingEncoding => (encoding, text) => string.IsNullOrEmpty(text) ? 0 : GetCachedEncoding(encoding).Encode(text).Count;
 
+    /// <summary>
+    /// Calculates the length of a chat message with the specified content and author role, using a provided length function.
+    /// </summary>
+    /// <param name="content">The content of the chat message.</param>
+    /// <param name="authorRole">The <see cref="AuthorRole"/> of the message.</param>
+    /// <param name="lengthFunction">A function to calculate the length of a string.</param>
+    /// <returns>The total length for the chat message.</returns>
+    public static int LengthChatMessage(string content, AuthorRole authorRole, Func<string, int> lengthFunction)
+        => InnerLengthChatMessage(content, null, authorRole, (s, _) => lengthFunction(s));
+
+    /// <summary>
+    /// Calculates the length of a chat message with the specified content, encoding and author role, using a provided length function with encoding.
+    /// </summary>
+    /// <param name="content">The content of the chat message.</param>
+    /// <param name="encoding">The name of the GptEncoding.</param>
+    /// <param name="authorRole">The <see cref="AuthorRole"/> of the message.</param>
+    /// <param name="lengthFunctionWithEncoding">A function to calculate the length of a string with encoding.</param>
+    /// <returns>The total length for the chat message.</returns>
+    public static int LengthChatMessageWithEncoding(string content, string encoding, AuthorRole authorRole, Func<string, string, int> lengthFunctionWithEncoding)
+        => InnerLengthChatMessage(content, encoding, authorRole, lengthFunctionWithEncoding);
+
+    /// <summary>
+    /// Internal method to calculate the length of a chat message with the specified content, encoding and author role, using a provided length function with encoding.
+    /// </summary>
+    /// <param name="content">The content of the chat message.</param>
+    /// <param name="encoding">The name of the GptEncoding.</param>
+    /// <param name="authorRole">The <see cref="AuthorRole"/> of the message.</param>
+    /// <param name="lengthFunction">A function to calculate the length of a string.</param>
+    /// <returns>The total length for the chat message.</returns>
+    private static int InnerLengthChatMessage(string content, string encoding, AuthorRole authorRole, Func<string, string, int> lengthFunction)
```

Comment:
I would consider simplifying the method by changing it to:
```
private static int InnerLengthChatMessage(string content, AuthorRole authorRole, Func<string, int> lengthFunction)
{
    var tokenCount = authorRole == AuthorRole.System ? lengthFunction("\n") : 0;
    return tokenCount + lengthFunction($"role:{authorRole.Label}") + lengthFunction($"content:{content}");
}
```
And the corresponding calls would be:
```
=> InnerLengthChatMessage(content, authorRole, lengthFunction);
=> InnerLengthChatMessage(content, authorRole, s => lengthFunctionWithEncoding(s, encoding));
```
This is a subjective suggestion, but I find it easier to read. However, it might be a personal preference. Marking it as resolved :)

Reply:
As long as it works...

Reply:
Changed! Very smart approach